### ISSUE TITLE
When logging is on, prompts for passwords get out of order - reopened

### ISSUE THREAD

**Issue by fseatonuc** [picodotdev/alis#198]
I did some more testing and I'm still having the same problem as reported in #196 .  

> Give a try to the fix. I replaced the two exec commands with this only one.
>
> exec > >(tee -a $FILE) 2> >(tee -a $FILE >&2)

It's not 100% of the time, though.  I wrote a small test script that only calls `init_log` and `read_passwords` and sometimes it works and sometimes the output gets mixed, so I'm pretty sure it's a timing issue caused by having two `tee` processes running in parallel, one for STDOUT and one for STDERR.  The `read -sp` command writes its prompt to STDERR while the other echo commands in your `read_passwords` function are writing to STDOUT so it's basically a race to see which `tee` process will output to the terminal first.  

I should note the I have commented out the `set -o xtrace` as I want logging, but without debugging.  I opened a separate issue (#197) for a "wishlist" to have the `set -o xtrace` commands be tied to a separate option (perhaps `DEBUG`?) instead of the `LOG` option.   

Possible solutions:

1) use a single `tee` process instead of two by redirecting both STDOUT and STDERR to the same process.  Not only does this resolve the issue, but it should be more efficient as well as less dangerous than having two tee processes attempting to append to the same file at the same time. e.g.:
```
exec &> >(tee -a $FILE)
```

2) Force the `read` commands to write their prompt to STDOUT instead of STDERR.  e.g.:
```
            read -sp 'Type LUKS password: ' LUKS_PASSWORD 2>&1
            read -sp 'Retype LUKS password: ' LUKS_PASSWORD_RETYPE 2>&1
            read -sp 'Type WIFI key: ' WIFI_KEY 2>&1
            read -sp 'Retype WIFI key: ' WIFI_KEY_RETYPE 2>&1
            read -sp 'Type root password: ' ROOT_PASSWORD 2>&1
            read -sp 'Retype root password: ' ROOT_PASSWORD_RETYPE 2>&1
            read -sp 'Type user password: ' USER_PASSWORD 2>&1
            read -sp 'Retype user password: ' USER_PASSWORD_RETYPE 2>&1
```

Both solutions are working for me, but solution 1 is more efficient.

**Comment by picodotdev** [comment]
I used option 1.

https://github.com/picodotdev/alis/blob/master/alis-commons.sh#L172
